src/common/foldernode.js

```diff
@@ -141,16 +141,54 @@ export class FolderNode extends BaseNode {
     return node;
   }
 
-  get fullNameParts() {
+  getFullNameParts(includeAccountNode) {
     let parts = [];
     let node = this; // eslint-disable-line consistent-this
-    while (node && !(node instanceof AccountNode)) {
-      parts.unshift(node.item.name);
+    while (node) {
+      let isAcctNode = node instanceof AccountNode;
+      if (includeAccountNode || !isAcctNode) {
+        parts.unshift(node.item.name);
+      }
+      if (isAcctNode) {
+        break;
+      }
       node = node.parent;
     }
 
     return parts;
   }
+  
+  get fullPathReversed() {
```

Comment:
I think we should separate the display and data aspects. We can get the reversed parts with `getFullNameParts(true).reverse()`, then in the folder nodes we can use templating and css to set the styles.

On the gmail and empty components thing, I think we could generally filter out empty components in getFullNameparts, and maybe add an option skipGmail that would then always skip [Gmail] folders.